### DISCUSSION TITLE
Adding player kill and death statistics

### DISCUSSION THREAD

**Discussion by Mrkillero4ka** [Zrips/CMI#7636] [Feature Requests]
Add new statistics to CMI as:
1. Killing players.
Example: If a player killed another player, the player who killed him was credited with the kill.
2. Death of a player.
Example: If a player died, then this was read to him.
Also add tops for them

**Comment by CloudeLecaw** [comment]
well at least kills and deaths are allready part of the statisics

```
#         Deaths - count of deaths
#         MobKills - total amount of mobs killed by player
#         MonsterKills - total amount of monsters killed by player
#         PlayerKills - total amount of players killed. Indirect kills dont count
```

These staticiscs can be shown by placeholders

> %cmi_user_stats_[mainStat(:optionalSubStat)]% returns players statics by provided criteria. The format is the same as used for rankup system. For example %cmi_user_stats_MonsterKills% returns the total number of monster kills, while %cmi_user_stats_MonsterKills:zombie% will return the total zombie kills.
> This placeholder can return 3 types of values, clean number, like for mob kills, time, like total playtime and distance. 

What we dont have this the Topslist for deaths and kills.

**Reply by Mrkillero4ka** [reply]
And then, you only need to make tops, I hope I will :)

**Reply by Mrkillero4ka** [reply]
Will tops be added?

**Reply by CloudeLecaw** [reply]
> Will tops be added?

Maybe if Zrips has time in a few years^^ So wait if he sees that he may has time
a request is never a promiss something will ever be added

Zrips time is limited and a also other things have other priority
So if he things he wanne add it he add it, if not then not.
May your request never got answered maybe it will. Noone can tell you this.

**Reply by Mrkillero4ka** [reply]
ok

**Reply by MrCrazys** [reply]
> Will tops be added?

You can just use a plugin like ajLeaderboards (https://www.spigotmc.org/resources/ajleaderboards.85548/) to generate placeholders for showing the top values for something. Seems a bit out of the scope of CMI.

**Reply by Mrkillero4ka** [reply]
The ajLeaderboards plugin does not update the statistics of the players, it updates it after a reboot, even if the CMI has tops by time and coins, then why can't you add tops by deaths and murders if there are their placeholders in the CMI

**Reply by MrCrazys** [reply]
> The ajLeaderboards plugin does not update the statistics of the players, it updates it after a reboot, even if the CMI has tops by time and coins, then why can't you add tops by deaths and murders if there are their placeholders in the CMI

It does update it. Works fine for me.

**Reply by Mrkillero4ka** [reply]
It does not update for me, more + if you changed your nickname through CMI, then in the tops from ajLeaderboards he will not change it in the top, and in the tops from CMI he will change the nickname even if it is fake

**Reply by MrCrazys** [reply]
> It does not update for me, more + if you changed your nickname through CMI, then in the tops from ajLeaderboards he will not change it in the top, and in the tops from CMI he will change the nickname even if it is fake

That doesn't seem like a CMI issue. As I said, it works fine for me. If it doesn't update, it isn't a CMI issue.

**Reply by Mrkillero4ka** [reply]
Personally, when I removed this plugin, the server began to work much faster and it can be seen, now with 70-80 players there are no lags

**Reply by mrfloris** [reply]
> Personally, when I removed this plugin, the server began to work much faster and it can be seen, now with 70-80 players there are no lags
This is unrelated to the suggestion, please stay on topic.

**Reply by Mrkillero4ka** [reply]
ok

**Comment by mrfloris** [comment]
let's not go around in circles. you're free to suggest ajleaderboard, and the other person is free to not use it. 

MrKIller: Thank you for the suggestion, it doesn't guarantee it will get added, or when. In the meantime, ajLeaderboard is an option - if you don't want to use it that's fine. You have to wait until cmi perhaps adds it in the future (if ever)

**Reply by Mrkillero4ka** [reply]
I have nothing against him I don’t know how, I just suggested an idea, and there how zrips mode is to add it or not
